src/components/form-elements/radio/_radio.scss

```diff
@@ -0,0 +1,128 @@
+$radioSize: 16px;
+$radioClickableArea: 32px;
+$radioPadding: 4px;
+$radioUncheckedBorderWidth: 2px;
+$radioCheckedBorderWidth: 5px;
+
+.sg-radio {
+  --radioColor: #{$black};
+  --radioHoverColor: #{$gray-60};
+  --radioInvalidColor: #{$red-60};
+  --radioInvalidHoverColor: #{$red-50};
+  --radioLabelColor: #{$text-black};
+  --radioDescriptionColor: #{$text-black};
+  --radioOutlineColor: #{$blue-40};
+
+  display: inline-flex;
+  flex-direction: column;
+  padding: $radioPadding 0;
+
+  &--light {
+    --radioColor: #{$white};
+    --radioHoverColor: #{$gray-40};
+    --radioInvalidColor: #{$red-40};
+    --radioInvalidHoverColor: #{$red-50};
+    --radioLabelColor: #{$text-white};
+    --radioDescriptionColor: #{$text-white};
+    --radioOutlineColor: #{$blue-40};
+  }
+
+  &--disabled {
+    opacity: 0.45;
+    cursor: not-allowed;
+    pointer-events: none;
+  }
+
+  &__wrapper {
+    display: flex;
+
+    &:hover {
+      .sg-radio__circle {
+        box-shadow: 0px 0px 0px $radioUncheckedBorderWidth
+          var(--radioHoverColor) inset;
+      }
+
+      .sg-radio__input {
+        &:checked + .sg-radio__circle {
+          box-shadow: 0px 0px 0px $radioCheckedBorderWidth
+            var(--radioHoverColor) inset;
+        }
+        &[aria-invalid] + .sg-radio__circle {
+          box-shadow: 0px 0px 0px $radioUncheckedBorderWidth
+            var(--radioInvalidHoverColor) inset;
+        }
+        &:checked[aria-invalid] + .sg-radio__circle {
+          box-shadow: 0px 0px 0px $radioCheckedBorderWidth
+            var(--radioInvalidHoverColor) inset;
+        }
+      }
+    }
+  }
+
+  &__element {
+    position: relative;
+    width: $radioClickableArea;
+    height: $radioClickableArea;
+    display: inline-flex;
+    flex-shrink: 0;
+    align-items: center;
+    justify-content: center;
+  }
+
+  &__input {
+    width: 100%;
+    height: 100%;
+    position: absolute;
+    opacity: 0;
+    top: 0;
+    left: 0;
+    z-index: 1;
+
+    &:focus + .sg-radio__circle {
+      outline: $radioUncheckedBorderWidth solid var(--radioOutlineColor);
+      outline-offset: 2px;
+    }
+
+    &:checked + .sg-radio__circle {
```

Comment:
Style for Windows High Contrast Mode is missing. `we can discussed that together :)